R/mzQC.R

```diff
@@ -1,1010 +1,1310 @@
+    fromData = function( data, context = "MzQCmetadata")
+    {
+      # Define expected fields
+      expected_fields <- c("label", "inputFiles", "analysisSoftware", "cvParameters")
+      
+      # Check for unexpected fields
+      checkUnexpectedFields(data, expected_fields, "MzQCmetadata", context)
+      
+      # Required fields
+      self$label = check_field_exists(data, "label", "MzQCmetadata", paste0(context, "$label"), NA_character_)
+      
+      inputFiles_data = check_field_exists(data, "inputFiles", "MzQCmetadata", paste0(context, "$inputFiles"), list())
+      self$inputFiles = fromDatatoMzQC(MzQCinputFile, inputFiles_data, context = paste0(context, "$inputFiles"))
+      
+      analysisSoftware_data = check_field_exists(data, "analysisSoftware", "MzQCmetadata", paste0(context, "$analysisSoftware"), list())
+      self$analysisSoftware = fromDatatoMzQC(MzQCanalysisSoftware, analysisSoftware_data, context = paste0(context, "$analysisSoftware"))
+      
+      # Optional cvParameters
+      cvParameters_data = getOptionalValue(data, "cvParameters", list())
+      if (length(cvParameters_data) > 0) {
+        self$cvParameters = fromDatatoMzQC(MzQCcvParameter, cvParameters_data, context = paste0(context, "$cvParameters"))
+      } else {
+        self$cvParameters = list()
+      }
+      
+      return(self)
+    }
+  )
+)
+setOldClass("MzQCmetadata")
+asJSON.MzQCmetadata = function(x, ...) x$toJSON(...)
+setMethod("asJSON", "MzQCmetadata", asJSON.MzQCmetadata)
+
+################################################################################################################################
+#################################################################################################################################'
+#' The central class to store QC information
+#'
+#' @field accession Accession number identifying the term within its controlled vocabulary.
+#' @field name Name of the controlled vocabulary element describing the metric.
+#' @field description (optional) Definition of the controlled vocabulary term.
+#' @field value (optional) Value of the metric (single value, n-tuple, table, matrix).
+#'        The structure is not checked by our mzQC implementation and must be handled by the caller, see \code{\link{toQCMetric}}.
+#' @field unit (optional) Array of unit(s), stored as MzQcvParameter
+#'
+#' @exportClass MzQCqualityMetric
+#' @export MzQCqualityMetric
+#'
+MzQCqualityMetric = R6Class(
+  'MzQCqualityMetric',
+  
+  # Options
+  portable = FALSE,
+  cloneable = FALSE,
+  
+  public = list(
+    
+    ## Fields
+    accession = 'character',
+    name = 'character',
+    description = 'character', # optional
+    value = 'ANY',             # optional value of unspecified type
+    unit = 'list',              # optional array of MzQCcvParameter
+
+    ## Methods
+    
+    #' @description
+    #' Constructor
+    #' @param accession Optional string value for initialization of field accession
```

Comment:
Delete `Optional ` if required (global, i.e. in this file).
Bonus: scratch (` = NA_character_`) where appropriate